internal/cmd/operator-sdk/alpha/config3alphato3/convert_config_3-alpha_to_3_test.go

```diff
@@ -0,0 +1,130 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config3alphato3
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/format"
+)
+
+var _ = Describe("ConvertConfig3AlphaTo3", func() {
+	TruncatedDiff = false
+	DescribeTable("should return the expected config",
+		func(inputCfgStr, expectedCfgStr string) {
+			output, err := convertConfig3AlphaTo3([]byte(inputCfgStr))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(output)).To(Equal(expectedCfgStr))
+		},
+		Entry("basic", basicConfig, basicConfigExp),
+		Entry("no resources", noResourcesConfig, noResourcesConfigExp),
+		Entry("multiple resources", multipleAPIsConfig, multipleAPIsConfigExp),
+	)
+})
+
+const (
+	basicConfig = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+resources:
+- crdVersion: v1
+  group: cache
+  kind: Memcached
+  version: v1alpha1
+version: 3-alpha
+`
+	basicConfigExp = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+resources:
+- api:
+    crdVersion: v1
+    # TODO(user): Uncomment the below line if this resource's CRD is namespace scoped, else delete it.
+    # namespaced: true
+  # TODO(user): Uncomment the below line if this resource implements a controller, else delete it.
+  # controller: true
+  domain: example.com
+  group: cache
+  kind: Memcached
+  version: v1alpha1
+version: "3"
+`
+)
+
+const (
+	noResourcesConfig = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+version: 3-alpha
+`
+	noResourcesConfigExp = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+version: "3"
+`
+)
+
+const (
+	multipleAPIsConfig = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+resources:
+- crdVersion: v1
+  group: cache
+  kind: Memcached
+  version: v1alpha1
+  webhookVersion: v1
+- crdVersion: v1
+  group: cache
+  kind: MemcachedRS
+  version: v1alpha1
+version: 3-alpha
+`
+	multipleAPIsConfigExp = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+resources:
+- api:
+    crdVersion: v1
+    # TODO(user): Uncomment the below line if this resource's CRD is namespace scoped, else delete it.
+    # namespaced: true
+  # TODO(user): Uncomment the below line if this resource implements a controller, else delete it.
+  # controller: true
+  domain: example.com
+  group: cache
+  kind: Memcached
+  version: v1alpha1
+  webhooks:
+    # TODO(user): Uncomment the below line if this resource's webhook implements a conversion webhook, else delete it.
```

Comment:
what happens if the user does not solve the todos?

Reply:
Nothing. This command writes a valid config.

Reply:
It's better than them not being able to use `operator-sdk` at all, which would be the case if they didn't run this command. They'd see an error like
```
FATA[0000] Version "3-alpha" not supported
```